packages/angular/projects/ui-angular/src/lib/components/authenticator/components/setup-totp/setup-totp.component.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { AuthenticatorService } from '../../../../services/authenticator.service';
+import { SetupTotpComponent } from './setup-totp.component';
+import { AmplifySlotComponent } from '../../../../utilities/amplify-slot/amplify-slot.component';
+import { ErrorComponent } from '../../../../primitives/error/error.component';
+import { BaseFormFieldsComponent } from '../base-form-fields/base-form-fields.component';
+import { FormFieldComponent } from '../form-field/form-field.component';
+import { ButtonComponent } from '../../../../primitives/button/button.component';
+import { TestBed, ComponentFixture } from '@angular/core/testing';
+import QRCode from 'qrcode';
+import { Auth } from 'aws-amplify';
+import { MockComponent } from 'ng-mocks';
+import { getTotpCode } from '@aws-amplify/ui';
+
+const mockUser = { username: 'username' };
+const mockContext = {
+  country_code: '100',
+  formFields: { setupTOTP: { QR: null } },
+  user: mockUser,
+};
+
+const DEFAULT_TOTP_ISSUER = 'AWSCognito';
+const SECRET_KEY = 'secretKey';
+
+const setupTOTPSpy = jest.spyOn(Auth, 'setupTOTP');
+
+const toDataURLSpy = jest.spyOn(QRCode, 'toDataURL');
+
+describe('SetupTotpComponent', () => {
+  let fixture: ComponentFixture<SetupTotpComponent>;
+  let component: SetupTotpComponent;
+
+  beforeEach(async () => {
+    jest.resetAllMocks();
+
+    setupTOTPSpy.mockResolvedValue(SECRET_KEY);
+
+    const mockAuthenticatorService = {
```

Comment:
Does this need to be recreated between each test? If not should be able to move this outside the scope of the top level `describe` block

Reply:
For the mock resolved value to resolve correctly it has to be in a async/await function. But to your point, it doesn't need to be run on every test. I'll move it to the test it needs to be in.

Reply:
b226db3